### ISSUE TITLE
Brave core dumps on Fedora 38 with nvidia drivers

### ISSUE THREAD

**Issue by pghpete** [brave/brave-browser#29871]
## Description 
Brave core dumps on Fedora 38 with the non-free nvidia drivers.  I have tried completely removing and reinstalling but unfortunately that didn't work.

## Steps to Reproduce
launch brave via icon or brave-browser on the command line

## Actual result:
/usr/bin/brave-browser: line 48: 15862 Illegal instruction     (core dumped) "$HERE/brave" "$@"

## Expected result:
Brave to open properly without core dumping.

## Reproduces how often: 
Happens every time

## Brave version (info found on brave://version)
Fedora 38
brave-browser.x86_64                                      1.50.121-1                         @brave-browser

## Version/Channel Information:
- Can you reproduce this issue with the current release?
yes

## Other Additional Information:

- Is the issue reproducible on the latest version of Chrome? 
no

**Comment by CodeDead** [comment]
Core dumps as well with AMD drivers.

**Comment by AydroPunk** [comment]
The same thing happens to me, with the non-free drivers from nvidia, brave does not open and I get exactly the same errors as @pghpete in the new Fedora 38.:(
![Captura desde 2023-04-19 17-34-59](https://user-images.githubusercontent.com/90808551/233221742-ceecb3dd-1bea-4992-89b2-4401838e0969.png)

**Comment by HarveyDanger** [comment]
Also experiencing the same issue, but with XE Iris on IdeaPad 5; Brave: Release Notes v1.50.121 (Apr 19, 2023)
Crashes on every tab with SIGILL.
```
Linux 6.2.11-300.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Apr 13 20:27:09 UTC 2023 x86_64 GNU/Linux
```

```
[8876:8876:0419/181526.239524:ERROR:chrome_browser_cloud_management_controller.cc(162)] Cloud management controller initialization aborted as CBCM is not enabled.
[8876:8876:0419/181529.522976:ERROR:object_proxy.cc(623)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not implemented
[8913:8913:0419/181530.841997:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[8913:8913:0419/181532.963630:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[8913:8913:0419/181532.975936:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
```

**Comment by Yawning** [comment]
FWIW this isn't limited to F38.  This also occurs on F37 (with AMD drivers, though the crash appears to be GPU vendor agnostic).

Upgrading to https://github.com/brave/brave-browser/releases/tag/v1.51.96 crashes.
Downgrading to https://github.com/brave/brave-browser/releases/tag/v1.50.120 works.

**Comment by pghpete** [comment]
@yawning I downgraded as well and mine worked to so it definitely appears to be an issue with 1.50.121-1 and not the OS or GPU. 1.50.119-1 works for me without issue

**Comment by pghpete** [comment]
Just as a test I removed Ver: 1.50-119-1 and disabled the brave repositories.

I then followed the instructions here: [https://brave.com/linux/#release-channel-installation](https://brave.com/linux/#release-channel-installation) to install the beta repos, import acs, and finally install brave-browser-beta Ver: 1.51.95-1 and it worked without issues.

I will roll back all of that and stick with the downgraded package version 1.50.119-1 for now however.

**Comment by Dskobra** [comment]
Having this exact issue on 1 computer. I got 2 computers a desktop and laptop both with amd dedicated gpus (cpu as well) and only the laptop was affected. Downgrading the package fixed it, but out of curiosity I updated brave after downgrading/restart and it also worked fine. So possibly some corruption with the package? O.o

**Comment by mostafaqanbaryan** [comment]
Same here

**Comment by mostafaqanbaryan** [comment]
Same here

**Comment by CodeDead** [comment]
Related issues:
https://github.com/brave/brave-browser/issues/29877
https://github.com/brave/brave-browser/issues/29878

It's not limited to Fedora 38 or Fedora in general.

**Comment by pitsi** [comment]
Can someone tell me where are those core dump files stored? I have lost like 500+mb of precious ssd storage from brave's crashes since this morning!

**Comment by CodeDead** [comment]
> Can someone tell me where are those core dump files stored? I have lost like 500+mb of precious ssd storage from brave's crashes since this morning!

brave://crashes/

You can clear them by clearing your Brave browser cache.

**Comment by pitsi** [comment]
Thank you for the reply, but...

FIrst and foremost, I have my brave's cache in /tmp (with `--disk-cache-dir=/tmp/cache`) and /tmp in ram and I have rebooted at least twice since this morning.
Second, I tried bleachbit (`bleachbit -c brave.cache`) because I hate opening menus and submenus, but it only deletes like 2mb.
Third, I tried brave://settings/clearBrowserData and it still mentions a small size like the 2mb above. As it seems, under the same menu and in the "on exit tab", I had set it to delete cache on exit.
Fourth, `brave://crashes` reports only one crash which happened ~30 minutes ago. This is probably the last one before I got really upset with the crashes today and I downgraded to 1.50.119.

And yet, those 500mb are still missing. Debian testing/unstable x64 is my distro.
I do not mean to hijack the conversation about the real issue here with my space issue, but any help is welcome.

**Comment by pghpete** [comment]
Mine were here...
/home/user/.config/BraveSoftware/Brave-Browser/Crash\ Reports/completed/

However mine were tiny.

**Comment by srirambv** [comment]
if there are any entries in `brave://crashes` please send the crash report or click on `Send more info` which will create a new issue in GH and will help us narrow down the issue.

**Comment by pghpete** [comment]
I have sent half a dozen or so. I don't have a "send more info" button or option.

**Comment by kjozwiak** [comment]
Firstly, I would like to apologies to anyone who's experiencing the above issues and appreciate all the reports and information. It looks like `112.0.5615.138` is causing issues on `Linux` and we suspect it might be due to https://bugs.chromium.org/p/chromium/issues/detail?id=1434194. What usually happens is when we get a chromium bump as per https://chromereleases.googleblog.com/2023/04/stable-channel-update-for-desktop_18.html, we update all our branches and then push the release on the targeted platforms. So for `112.0.5615.138`, both `macOS` & `Windows` were targeted. The update mentions that a `Linux` update will be coming soon. However, when we usually get bumps with zero-day vulnerabilities, we try to update all of our platforms ASAP. So we ended up pushing `112.0.5615.138` on `Linux` as well to help protect our users. However, in hindsight, we probably should have waited.

We're preparing a new update that will basically bump the version from `1.50.121` and revert users to `1.50.119` which was using `112.0.5615.121`. We've received a bunch of confirmations spanning various threads on both reddit and GitHub that reverting to `112.0.5615.121` fixes the issues. We've also took a look at the difference between `112.0.5615.138` & `112.0.5615.121` and see nothing that would cause any issues when reverting. We should have an update available within ~2-3hrs. In the meantime, I would advice `Linux` users not to update to `112.0.5615.138` and wait for the next chromium bump.

We'll also update the various Brave communication channels including reddit and http://community.brave.com to let users know what happened and let them know that an update is on the way that should restore functionality back to normal.

**Comment by pghpete** [comment]
@kjozwiak Thank you for the quick and through response! It is greatly appreciated! If you need testers on various linux platforms at any time, I'd be willing to help.

**Comment by kjozwiak** [comment]
> Version 1.50.125 Chromium: 112.0.5615.165 fixed the issue 👍

We pushed out `1.50.125 Chromium: 112.0.5615.165` which should fix the issue for everyone. If you were having issues, please let us know if the above release fixed your issue. You can also download the `deb` files manually via https://github.com/brave/brave-browser/releases/tag/v1.50.125. Just for some context, the above was fixed by upgrading chromium from `112.0.5615.138` to `112.0.5615.165`.
